cypress/integration/chi-js/side-nav.spec.js

```diff
@@ -0,0 +1,226 @@
+        cy.get(SIDENAV_COMPONENT_SELECTOR).within(() => {
+          cy.get('.chi-sidenav__list li')
+            .contains(menuText)
+            .as('firstLevelElement');
+
+          cy.get('@firstLevelElement')
+            .should('have.attr', 'href')
+            .then(href => {
+              cy.get('' + href).as('drawer');
+              cy.get('@firstLevelElement')
+                .click()
+                .wait(550);
+              cy.get(`${href}.-active`)
+                .should('be.visible')
+                .within(() => {
+                  cy.get('.-icon.-close')
+                    .click()
+                    .parent('.chi-drawer')
+                    .should('not.have.class', '-active');
+                });
+            });
+        });
+
+      });
+
+    });
+
+  });
+
+  describe('check that click on 1st level opens the drawer and click on empty space in menu container closes it', () => {
+    [/^Active Item$/, /^Menu Item$/].forEach(menuText => {
+      it(`check "${menuText}" clicked outside drawer closes it`, () => {
+        cy.get(SIDENAV_COMPONENT_SELECTOR).within(() => {
+          cy.get('.chi-sidenav__list li')
+            .contains(menuText)
+            .as('firstLevelElement');
+
+          cy.get('@firstLevelElement')
+            .should('have.attr', 'href')
+            .then(href => {
+              cy.get('' + href).as('drawer');
+              cy.get('@firstLevelElement')
+                .click()
+                .wait(550);
+              cy.get(`${href}.-active`).should('be.visible');
+              cy.get('.chi-sidenav__content')
+                .click()
+                .wait(250);
+              cy.get('' + href)
+                .should('not.have.class', '-active')
+                .should('not.be.visible');
+            });
+        });
+
+      });
+
+    });
+
+  });
+
+  describe('Click on 2nd level menus should open 3rd level drop-downs', () => {
+    [/^Active Item$/, /^Menu Item$/].forEach(menuText => {
+      var dropdownItems = [];
+      if (menuText == '/^Active Item$/') {
+        dropdownItems = ['Title A', 'Title B'];
+      } else {
+        dropdownItems = ['Title E', 'Title F'];
+      }
+      dropdownItems.forEach(element => {
+        it(`check "${menuText}" >   "${element}" should open Dropdown`, () => {
+          cy.get(SIDENAV_COMPONENT_SELECTOR).within(() => {
+            cy.get('.chi-sidenav__list li')
+              .contains(menuText)
+              .as('firstLevelElement');
+            cy.get('@firstLevelElement')
+              .should('have.attr', 'href')
+              .then(href => {
+                cy.get('' + href).as('drawer');
+                cy.get('@firstLevelElement')
+                  .click()
+                  .wait(550);
+                cy.get(`${href}.-active`).should('be.visible');
+                cy.get(
+                  href + ' .chi-sidenav__drawer-list li div.chi-sidenav__title'
```

Comment:
Since this kind of complex selectors are repeated in several tests, can we define a constant con top of the file and store the selector string there?
Something like:
`const drawerLiTitleSelector = ' .chi-sidenav__drawer-list li div.chi-sidenav__title';`
and inside tests:
`cy.get(href + drawerLiTitleSelector)`
The best practice would be to abstract all the selectors like this. It makes the code more readable.